Removed deprecated code from the following:-

simon-mundy opened the pull request:
Laminas\Db\Adapter\Adapter.php
- removed $lastPreparedStatement property
- removed createDriverFromParameters() method
- removed createPlatformFromDriver() method

Laminas\Db\Adapter\Driver\Oci8\Statement
- Removed $parametersBound property

Laminas\Db\Metadata\Metadata
- Removed entire file

Laminas\Db\Sql\Expression
- Removed $types property
- Removed $types param on constructor
- Removed set/get Types methods
- Updated unit tests to reflect changes

Laminas\Db\Sql\Select
- Removed JOIN_OUTER_LEFT and JOIN_OUTER_RIGHT constants

Laminas\Db\Sql\Sql
- Removed $sqlPlatform from constructor
- Removed getSqlStringForSqlObject() method
- Updated unit tests to reflect changes

Laminas\Db\Sql\TableIdentifier
- Removed setTable() method
- Removed setSchema() method
- Updated unit tests to reflect changes Changes to documentation to reflect correct Expression parameter usage

simon-mundy committed:
Removed deprecated code from the following:-

Laminas\Db\Adapter\Adapter.php
- removed $lastPreparedStatement property
- removed createDriverFromParameters() method
- removed createPlatformFromDriver() method

Laminas\Db\Adapter\Driver\Oci8\Statement
- Removed $parametersBound property

Laminas\Db\Metadata\Metadata
- Removed entire file

Laminas\Db\Sql\Expression
- Removed $types property
- Removed $types param on constructor
- Removed set/get Types methods
- Updated unit tests to reflect changes

Laminas\Db\Sql\Select
- Removed JOIN_OUTER_LEFT and JOIN_OUTER_RIGHT constants

Laminas\Db\Sql\Sql
- Removed $sqlPlatform from constructor
- Removed getSqlStringForSqlObject() method
- Updated unit tests to reflect changes

Laminas\Db\Sql\TableIdentifier
- Removed setTable() method
- Removed setSchema() method
- Updated unit tests to reflect changes
Changes to documentation to reflect correct Expression parameter usage

simon-mundy committed:
Removed deprecated code from the following:-

Laminas\Db\Adapter\Adapter.php
- removed $lastPreparedStatement property
- removed createDriverFromParameters() method
- removed createPlatformFromDriver() method

Laminas\Db\Adapter\Driver\Oci8\Statement
- Removed $parametersBound property

Laminas\Db\Metadata\Metadata
- Removed entire file

Laminas\Db\Sql\Expression
- Removed $types property
- Removed $types param on constructor
- Removed set/get Types methods
- Updated unit tests to reflect changes

Laminas\Db\Sql\Select
- Removed JOIN_OUTER_LEFT and JOIN_OUTER_RIGHT constants

Laminas\Db\Sql\Sql
- Removed $sqlPlatform from constructor
- Removed getSqlStringForSqlObject() method
- Updated unit tests to reflect changes

Laminas\Db\Sql\TableIdentifier
- Removed setTable() method
- Removed setSchema() method
- Updated unit tests to reflect changes
Changes to documentation to reflect correct Expression parameter usage

tyrsson commented on src/Adapter/Adapter.php:
```diff
@@ -2,15 +2,17 @@
-use InvalidArgumentException;
+use Exception as PhpException;
+use Laminas\Db\Adapter\Driver\ResultInterface;
+use Laminas\Db\Adapter\Profiler\ProfilerInterface;
```
If were going to use relative namespacing lets remove these.

simon-mundy committed:
Removed deprecated code from the following:-

Laminas\Db\Adapter\Adapter.php
- removed $lastPreparedStatement property
- removed createDriverFromParameters() method
- removed createPlatformFromDriver() method

Laminas\Db\Adapter\Driver\Oci8\Statement
- Removed $parametersBound property

Laminas\Db\Metadata\Metadata
- Removed entire file

Laminas\Db\Sql\Expression
- Removed $types property
- Removed $types param on constructor
- Removed set/get Types methods
- Updated unit tests to reflect changes

Laminas\Db\Sql\Select
- Removed JOIN_OUTER_LEFT and JOIN_OUTER_RIGHT constants

Laminas\Db\Sql\Sql
- Removed $sqlPlatform from constructor
- Removed getSqlStringForSqlObject() method
- Updated unit tests to reflect changes

Laminas\Db\Sql\TableIdentifier
- Removed setTable() method
- Removed setSchema() method
- Updated unit tests to reflect changes
Changes to documentation to reflect correct Expression parameter usage

simon-mundy committed:
Removed deprecated code from the following:-

Laminas\Db\Adapter\Adapter.php
- removed $lastPreparedStatement property
- removed createDriverFromParameters() method
- removed createPlatformFromDriver() method

Laminas\Db\Adapter\Driver\Oci8\Statement
- Removed $parametersBound property

Laminas\Db\Metadata\Metadata
- Removed entire file

Laminas\Db\Sql\Expression
- Removed $types property
- Removed $types param on constructor
- Removed set/get Types methods
- Updated unit tests to reflect changes

Laminas\Db\Sql\Select
- Removed JOIN_OUTER_LEFT and JOIN_OUTER_RIGHT constants

Laminas\Db\Sql\Sql
- Removed $sqlPlatform from constructor
- Removed getSqlStringForSqlObject() method
- Updated unit tests to reflect changes

Laminas\Db\Sql\TableIdentifier
- Removed setTable() method
- Removed setSchema() method
- Updated unit tests to reflect changes
Changes to documentation to reflect correct Expression parameter usage

tyrsson commented on src/Sql/Sql.php:
```diff
@@ -4,174 +4,121 @@
 use Laminas\Db\Adapter\AdapterInterface;
 use Laminas\Db\Adapter\Driver\StatementInterface;
-use Laminas\Db\Adapter\Platform\PlatformInterface;
+use Laminas\Db\Adapter\StatementContainerInterface;
```
I think we still need
use Laminas\Db\Adapter\Platform\PlatformInterface;
see line 17

tyrsson replied:
actually looks like we just need
use Laminas\Db\Adapter\Platform; ?

tyrsson commented on src/Sql/Sql.php:
```diff
@@ -4,174 +4,121 @@
             ));
         }
+
         return new Update($table ?: $this->table);
```
Might want to move to ?? ?
